### PR TITLE
warn users of mTLS certification expiration

### PR DESCRIPTION
**Pull request by tanvigour**
# Description

Issue warning for mTLS expiration

## Issue reference

Please reference the issue this PR will close: #3940

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/sentry/server/server.go [hunk 1]
```diff
@@ -102,6 +102,10 @@ func (s *server) getServerCertificate() (*tls.Certificate, error) {
 	issuerExp := s.certAuth.GetCACertBundle().GetIssuerCertExpiry()
 	serverCertTTL := issuerExp.Sub(now)
 
+	if serverCertTTL <= 30 {
```
Comment (yaron2) [inline pkg/sentry/server/server.go:105]: What is `30` here? is it supposed to be days? If it's days, then:

1. It won't work as the value of `time.Duration` doesn't represent days
2. the error message should be "The root CA cert is about to expire in %s days"

**Review by yaron2 — CHANGES_REQUESTED** [review]

File: pkg/sentry/server/server.go (hunk 1 above)
Comment (yaron2) [reply pkg/sentry/server/server.go:105]: Also, please add a test for this.

File: pkg/sentry/server/server.go [hunk 2]
```diff
@@ -101,6 +101,11 @@ func (s *server) getServerCertificate() (*tls.Certificate, error) {
 	now := time.Now().UTC()
 	issuerExp := s.certAuth.GetCACertBundle().GetIssuerCertExpiry()
 	serverCertTTL := issuerExp.Sub(now)
+	hoursLeft := serverCertTTL.Hours()
+
+	if serverCertTTL.Hours() <= 720 {
+		log.Warn("The root CA cert is about to expire in %s days", hoursLeft/24)
+	}
```
Comment (Taction) [inline pkg/sentry/server/server.go:108]: Some minor changes. You should use `Warnf` here, and `%f` to format this float value.
```
        if hoursLeft := serverCertTTL.Hours(); hoursLeft <= 720 {
		log.Warnf("The root CA cert is about to expire in %f days", hoursLeft/24)
	}
```

**Comment by Taction** [comment]
@tanvigour Also please fix DCO issues.

**Comment by berndverst** [comment]
I think we can abandon this PR in favor of https://github.com/dapr/dapr/pull/4328 which was merged. The proper way to be warned is to watch Sentry logs and enable alerting based on the Sentry logs.

Also there are other changes in the CLI that provide additional warnings there.

@artursouza @daixiang0 @yaron2 I recommend closing this PR here.

**Pull request closed without merging**